### PR TITLE
Add WalletLink support for Coinbase wallet

### PR DESCRIPTION
**Pull request by redben**
The WalletLink API is half-baked, but we need to integrate it to support Coinbase mobile app. Since the Coinbase mobile app doesn't support BSC, nor provides a way to specify a custom chain config this wallet only needs to be supported for Ethereum logins.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -87,6 +87,8 @@ export default new Vue({
     } else if (window.ethereum) { // modern MetaMask-like wallet
       // @ts-ignore
       const ethereum = window.ethereum
+      // walletlink steals typing of window.ethereum
+      // @ts-ignore
```
Comment (enlight) [inline src/index.ts:91]: Hmm, so actually... I don't think we want any of this auto-select wallet stuff to run at all when `features.bscWallets === true`

File: src/store/ethereum/wallets/walletlink.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        const wlProvider = walletLink.makeWeb3Provider(rpcUrl, chainId)
+
+        let accounts: string[] = []
+        try {
+            accounts = await wlProvider.send("eth_requestAccounts")
+        } catch (err) {
+            if (err.code === 4001) {
+                // EIP-1193 userRejectedRequest error
+                // If this happens, the user rejected the connection request.
+                console.log("Use rejected WalletLink connection request")
```
Comment (enlight) [inline src/store/ethereum/wallets/walletlink.ts:41]:
```suggestion
                console.log("User rejected WalletLink connection request")
```

File: src/store/ethereum/wallets/walletlink.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    mobile: true,
+    detect() {
+        return true
+    },
+    async createProvider(config: EthereumConfig): Promise<IWalletProvider> {
+        const chainId = parseInt(config.networkId, 10)
+        // FIXME: Hardcoded to mainnet right now since WalletLink API is half-baked and doesn't provide
+        // a way for the user to specify the network within the wallet app.
+        if (chainId !== 1) {
+            alert("WalletLink only works with mainnet for the moment " + config.networkId)
```
Comment (enlight) [inline src/store/ethereum/wallets/walletlink.ts:22]: It does let me pick a different network in the Coinbase wallet app... you sure this doesn't work with other chain IDs?

File: src/store/ethereum/wallets/walletlink.ts (hunk 2 above)
Comment (redben) [reply src/store/ethereum/wallets/walletlink.ts]: Fixed

File: src/store/ethereum/wallets/walletlink.ts (hunk 3 above)
Comment (redben) [reply src/store/ethereum/wallets/walletlink.ts]: Removed

File: src/index.ts (hunk 1 above)
Comment (redben) [reply src/index.ts:91]: Let's make deal with that in another PR

Comment (redben) [reply src/index.ts:91]: removed

File: src/views/FirstPage.vue [hunk 4]
```diff
@@ -388,7 +503,7 @@ export default class FirstPage extends Vue {
   }
 
   get bscWalletsEnabled(): boolean {
-    return this.$state.activeConfig ? this.$state.activeConfig.features.bscWallets : false
+    return this.$state.env ? this.$state.activeConfig!.features.bscWallets : false
```
Comment (enlight) [inline src/views/FirstPage.vue:506]: This was a null/undefined check before... why the change?

File: src/views/FirstPage.vue [hunk 5]
```diff
@@ -388,7 +503,7 @@ export default class FirstPage extends Vue {
   }
 
   get bscWalletsEnabled(): boolean {
-    return this.$state.activeConfig ? this.$state.activeConfig.features.bscWallets : false
+    return this.$state.activeConfig ? this.$state.activeConfig!.features.bscWallets : false
```
Comment (enlight) [inline src/views/FirstPage.vue:506]:
```suggestion
    return this.$state.activeConfig ? this.$state.activeConfig.features.bscWallets : false
```

**Review by enlight — COMMENTED** [review]
LGTM
